src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -109,6 +109,10 @@ internal static unsafe nuint GetIndexOfFirstNonAsciiByte(byte* pBuffer, nuint bu
             {
                 return GetIndexOfFirstNonAsciiByte_Intrinsified(pBuffer, bufferLength);
             }
+            else if (Vector128.IsHardwareAccelerated)
```

Comment:
The above `Sse2` or `AdvSimd` paths are the code paths that use `Vector128` and are much more efficient than the associated `Vector128` path.

I believe if it reaches this point, where we cannot use `Vector512`, `Vector256`, or `Sse2/AdvSimd`, then the `Vector` path itself will likely be as efficient and no need to call back into `_Vector`.

Reply:
We would prefer the Vector128 path for wasm

Reply:
Ok, understood. Thanks for the context.